test/edgehog_device_forwarder/monitor_test.exs

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+defmodule EdgehogDeviceForwarder.MonitorTest do
+  use EdgehogDeviceForwarder.CacheCase
+  alias EdgehogDeviceForwarder.Monitor
+
+  setup do
+    process = spawn(fn -> until_exit_process() end)
+    [process: process]
+  end
+
+  test "callback closure is executed once the process exits", %{process: process} do
+    message = :done
+    me = self()
+    send_message_to_self = fn -> send(me, message) end
+    Monitor.monitor(process, send_message_to_self)
+
+    send(process, :exit)
+    assert_receive ^message
+  end
+
+  def until_exit_process do
+    receive do
+      :exit -> :ok
+      _ -> until_exit_process()
```

Comment:
Receive blocks if it doesn't find any messages matching the pattern, so you can remove this and avoid the process spinning endlessly.
Also, I'd call this `wait_for_exit`

Reply:
But I want it to block, as I need to control _when_ the process has to finish